src/main/java/controller/EngineController.java

```diff
@@ -0,0 +1,182 @@
+                        case 2 -> getAllEngine();
+                        case 3 -> updateEngine();
+                        case 4 -> deleteEngine();
+                        case 5 -> {
+                            return;
+                        }
+                        default -> System.out.println("Entered incorrect data");
+                    }
+                } else {
+                    return;
+                }
+            }
+        });
+    }
+
+
+    private FuelType chooseFuelType() {
+        System.out.println("\nChoose a fuel type:\n" +
+                "1 - Petrol\n2 - Diesel\n" +
+                "3 - Gas\n4 - Electricity");
+        int choose = Integer.parseInt(scanner.nextLine());
+        return switch (choose) {
+            case 1 -> fuelTypeService.getByName(EngineCharacteristics.PETROL);
+            case 2 -> fuelTypeService.getByName(EngineCharacteristics.DIESEL);
+            case 3 -> fuelTypeService.getByName(EngineCharacteristics.GAS);
+            case 4 -> fuelTypeService.getByName(EngineCharacteristics.ELECTRICITY);
+            default -> throw new CarRentalException("You entered incorrect data");
+        };
+    }
+
+
+    private TransmissionType chooseTransmissionType() {
+        System.out.println("\nChoose a transmission type:\n" +
+                "1 - Automatic\n2 - Mechanic");
+        int choose = Integer.parseInt(scanner.nextLine());
+        return switch (choose) {
+            case 1 -> transmissionTypeService.getByName(EngineCharacteristics.AUTOMATIC);
+            case 2 -> transmissionTypeService.getByName(EngineCharacteristics.MECHANIC);
+            default -> throw new CarRentalException("You entered incorrect data");
+        };
+    }
+
+
+    public void insertEngine() {
+        handleException(() -> {
+            System.out.println("Enter the max speed:");
+            int maxSpeed = Integer.parseInt(scanner.nextLine());
+
+            FuelType fuelType = chooseFuelType();
+            TransmissionType transmissionType = chooseTransmissionType();
+
+            System.out.println("Enter the volume:");
+            double volume = Double.parseDouble(scanner.nextLine());
+
+            System.out.println("Enter the fuel consumption:");
+            double fuelConsumption = Double.parseDouble(scanner.nextLine());
+
+            Engine insertedEngine = engineService.insert(Engine.builder()
+                    .maxSpeed(maxSpeed)
+                    .fuelTypeId(fuelType.getId())
+                    .transmissionTypeId(transmissionType.getId())
+                    .volume(volume)
+                    .fuelConsumption(fuelConsumption)
+                    .build());
+            System.out.println("Engine has been inserted successfully");
+            System.out.println(insertedEngine.toString());
+        });
+    }
+
+
+    private void getAllEngine() {
+        handleException(() -> engineService.getAll().forEach(System.out::println));
```

Comment:
Could you please rename this method into plural form, add the counter in a loop, and check if there are no engines